rpc/handlers.go

```diff
@@ -72,20 +342,679 @@ func (s *rpcServer) GetV1QueryAllChainParams(ctx echo.Context) error {
+	currentHeight := int64(s.GetBus().GetConsensusModule().CurrentHeight())
+	if currentHeight > 0 {
+		currentHeight -= 1
+	}
+	height := body.Height
+	if height == 0 {
+		height = currentHeight
+	}
+	readCtx, err := s.GetBus().GetPersistenceModule().NewReadContext(currentHeight)
+	if err != nil {
+		return ctx.String(http.StatusInternalServerError, err.Error())
+	}
+	allFishermen, err := readCtx.GetAllFishermen(height)
+	if err != nil {
+		return ctx.String(http.StatusInternalServerError, err.Error())
+	}
+
+	start, end, totalPages, err := getPageIndexes(len(allFishermen), int(body.Page), int(body.PerPage))
+	if err != nil && !errors.Is(err, errNoItems) {
+		return ctx.String(http.StatusBadRequest, err.Error())
+	}
+	if totalPages == 0 || errors.Is(err, errNoItems) {
+		return ctx.JSON(http.StatusOK, QueryFishermenResponse{})
+	}
+
+	rpcFishermen := make([]ProtocolActor, 0)
+	for _, fm := range allFishermen[start : end+1] {
+		actor := protocolActorToRPCProtocolActor(fm)
+		rpcFishermen = append(rpcFishermen, actor)
+	}
+
+	return ctx.JSON(http.StatusOK, QueryFishermenResponse{
+		Fishermen:      rpcFishermen,
+		TotalFishermen: int64(len(allFishermen)),
+		Page:           body.Page,
+		TotalPages:     int64(totalPages),
+	})
+}
+
+func (s *rpcServer) GetV1QueryHeight(ctx echo.Context) error {
+	// Get latest stored block height
+	currentHeight := s.GetBus().GetConsensusModule().CurrentHeight()
+	if currentHeight > 0 {
+		currentHeight -= 1
+	}
+
+	return ctx.JSON(http.StatusOK, QueryHeight{
+		Height: int64(currentHeight),
+	})
+}
+
+func (s *rpcServer) PostV1QueryParam(ctx echo.Context) error {
+	var body QueryParameter
+	if err := ctx.Bind(&body); err != nil {
+		return ctx.String(http.StatusBadRequest, "bad request")
+	}
+
+	// Get latest stored block height
+	currentHeight := int64(s.GetBus().GetConsensusModule().CurrentHeight())
+	if currentHeight > 0 {
+		currentHeight -= 1
+	}
+	height := body.Height
+	if height == 0 || height > currentHeight {
+		height = currentHeight
+	}
+
+	readCtx, err := s.GetBus().GetPersistenceModule().NewReadContext(height)
+	if err != nil {
+		return ctx.String(http.StatusInternalServerError, err.Error())
+	}
+	paramValue, err := readCtx.GetStringParam(body.ParamName, height)
+	if err != nil {
+		return ctx.String(http.StatusInternalServerError, err.Error())
+	}
+
+	return ctx.JSON(http.StatusOK, Parameter{
+		ParameterName:  body.ParamName,
+		ParameterValue: paramValue,
+	})
+}
+
+func (s *rpcServer) PostV1QueryServicer(ctx echo.Context) error {
+	var body QueryAddressHeight
+	if err := ctx.Bind(&body); err != nil {
+		return ctx.String(http.StatusBadRequest, "bad request")
+	}
+
+	// Get latest stored block height
+	currentHeight := int64(s.GetBus().GetConsensusModule().CurrentHeight())
+	if currentHeight > 0 {
+		currentHeight -= 1
+	}
+	height := body.Height
+	if height == 0 {
+		height = currentHeight
+	}
+	readCtx, err := s.GetBus().GetPersistenceModule().NewReadContext(currentHeight)
```

Comment:
Should this not take in `height` instead of `currentHeight`?

If that's the case, I think we can also avoid needing to have the `CurretnHeight` call unless `height == 0`